test/parse_request.cc

```diff
@@ -254,8 +254,8 @@ std::string get_request_str(const std::string& grandparent_key,
                             const std::string& parent_key,
                             const std::string& key,
                             const bool specified_value) {
-  return R"({")" + grandparent_key + R"(":{")" + parent_key + R"(":{")" + key + R"(":)" +
-         std::string(specified_value ? "true" : "false") + R"(}}})";
+  return R"({"costing":")" + parent_key + R"(",")" + grandparent_key + R"(":{")" + parent_key +
```

Comment:
jeez, this whole test file is an abomination hahaha

Reply:
its jsons fault really. c++ needs a way like python where you can do format strings as block literals:

```python
f'''
{"i am": "a literal with variable {1+2} substitution"}
'''
```

Reply:
urgh yeah.. but it's also those 100s of helper functions with the generally annoying habit of not failing tests in an obvious way and you have to start the debugger to know which line really failed. personally I take a 10k LOC test file repeating itself over and over instead of a 1k LOC test file which abstracts too much